genesis/engine/sensors/camera.py

```diff
@@ -0,0 +1,996 @@
+        self._options: RasterizerCameraOptions
+        self._camera_node = None
+        self._camera_target = None
+        self._attached_link = None
+        self._attached_offset_T = None
+
+    # ========================== Sensor Lifecycle ==========================
+
+    def build(self):
+        """Initialize the rasterizer and register this camera."""
+        super().build()
+
+        scene = self._manager._sim.scene
+
+        # Initialize shared metadata on first camera
+        if self._shared_metadata.sensors is None:
+            self._shared_metadata.sensors = []
+            self._shared_metadata.lights = gs.List()
+            self._shared_metadata.camera_nodes = {}
+            self._shared_metadata.camera_targets = {}
+            self._shared_metadata.image_cache = {}
+
+            # Create standalone rasterizer context
+            self._shared_metadata.context = self._create_standalone_context(scene)
+
+            # Create offscreen rasterizer
+            from genesis.vis.rasterizer import Rasterizer
+
+            self._shared_metadata.renderer = Rasterizer(viewer=None, context=self._shared_metadata.context)
+            self._shared_metadata.renderer.build()
+
+        # Register this camera
+        self._shared_metadata.sensors.append(self)
+
+        # Add camera to rasterizer
+        self._add_camera_to_rasterizer()
+
+        # Initialize image cache for this camera
+        n_envs = max(self._manager._sim._B, 1)
+        h, w = self._options.res[1], self._options.res[0]
+        self._shared_metadata.image_cache[self._idx] = np.zeros((n_envs, h, w, 3), dtype=np.uint8)
+
+    def _create_standalone_context(self, scene):
+        """Create a simplified RasterizerContext for camera sensors."""
+        from genesis.vis.rasterizer_context import RasterizerContext
+        from genesis.options.vis import VisOptions
+
+        # Create minimal visualizer options
+        vis_options = VisOptions(
+            show_world_frame=False,
+            show_link_frame=False,
+            show_cameras=False,
+            rendered_envs_idx=list(range(max(self._manager._sim._B, 1))),
+        )
+
+        context = RasterizerContext(vis_options)
+        context.build(scene)
+        context.reset()
+        return context
+
+    def _add_camera_to_rasterizer(self):
+        """Add this camera to the rasterizer."""
+        # Use the rasterizer's add_camera method
+        camera_wrapper = self._get_camera_wrapper()
+        self._shared_metadata.renderer.add_camera(camera_wrapper)
+
+        # Set initial camera pose
+        self._update_camera_pose()
+
+    def _update_camera_pose(self):
+        """Update camera pose based on options."""
+        pos = torch.tensor(self._options.pos, dtype=gs.tc_float, device=gs.device)
+        lookat = torch.tensor(self._options.lookat, dtype=gs.tc_float, device=gs.device)
+        up = torch.tensor(self._options.up, dtype=gs.tc_float, device=gs.device)
```

Comment:
How does this work? It is done repeatedly during simulation? If so, avoid allocating memory on the fly.